contracts/BatchAuction.sol

```diff
@@ -0,0 +1,24 @@
+pragma solidity ^0.5.0;
+
+
+contract BatchAuction {
+
+    uint16 public maxAccountNumber = 10000;
+
+    mapping (address => uint16) public publicKeyToAccountMap;
+    mapping (uint16 => address) public accountToPublicKeyMap;
+
+    event AccountOpened(uint16 indexed depositor);
+
+    function openAccount(uint16 accountId) public {
+        require(accountId > 0, "Account index must be positive!");
```

Comment:
a uint is always positive (unsigned). I assume you want to prevent 0 from being a valid account (would write this as `accountId != 0`).
That's an interesting limitation that we haven't discussed yet (the snark does have an accountID 0). @josojo what do you think? Account0 could implicitly be the account of the contract owner and could be used e.g. to collect fees inside the snark.

Reply:
I like this idea very much. I could make this happen in the constructor which brings me to the question of who the owner should/could be. Either the owner is directly specified in the constructor parameters, or the deployer becomes the owner (i.e. msg.sender). Suggestions?

Should I implement accountID  at 0 in this PR?

Reply:
Warning; if we do use accountmap at index 0 to be that of the owner, we must be careful to recall that all addresses that do not occupy an account slot will have their inverse map evaluate to 0.